x-pack/plugins/security_solution/public/timelines/components/timeline/search_or_filter/index.tsx

```diff
@@ -110,19 +112,48 @@ const StatefulSearchOrFilterComponent = React.memo<Props>(
 
     const arrDataView = useMemo(() => (dataView != null ? [dataView] : []), [dataView]);
 
+    // Keep filter manager in sync with redux filters
```

Comment:
The synchronization logic used to live in a different component. I moved it up in the tree to reduce prop drilling and callback bubbling.

However, I'm wondering if this logic should rather live in a middleware since it's business logic that is rather unrelated to the component.

Reply:
Thanks for moving this up. And yea 100% it should be in middleware. We probably should have a timeline middleware library at this point 😅

Reply:
I'll keep it here for now and will move the business logic in a separate PR.